libbeat/publisher/pipeline/pipeline.go

```diff
@@ -154,13 +157,19 @@ func New(
 		monitors.Logger = logp.NewLogger("publish")
 	}
 
+	tracer, err := apm.NewTracer(beat.Beat, beat.Version)
+	if err != nil {
+		panic(err)
+	}
```

Comment:
Would it be possible to pass the tracer to the constructor? Is possible I'd prefer to not have a 'SetTracer' and 'getTracer' method. Maybe add it to the 'Monitors' type?